documentation/3-streams.md

```diff
@@ -164,6 +164,37 @@ Whether the socket was used for other previous requests.
 
 ## Events
 
+### `stream.on('response', ...)`
+
+#### `response`
+
+**Type: [`PlainResponse`](typescript.md#plainresponse)**
+
+This is emitted when a HTTP response is received.
+
+```js
+import {pipeline} from 'node:stream/promises';
+import {createWriteStream} from 'node:fs';
+import got from 'got';
+
+const readStream = got.stream('http://example.com/image.png', {throwHttpErrors: false});
+
+readStream.on('response', async response => {
```

Comment:
Using async function for an event listener is an anti-pattern as any thrown error will become an unhandled error. Instead, wrap the async stuff in an async-IIFE and use try/catch to properly handle any error.

Reply:
If there's a missing try/catch, there will be an unhandled error regardless whether it's an async function or an async IIFE.

Reply:
Hence, why I'm saying there should be a try/catch here. I think only `pipeline` could throw here, so it should be:

```js
(async () => {
	try {
		await pipeline(
	 		readStream,
	 		createWriteStream('image.png')
	 	);
	} catch {
		// handle it
	}
})();
```

Right?

Reply:
This can be one way. Another is:

```js
readStream.on('response', async response => {
	try {
		  await pipeline(
			  readStream,
			  createWriteStream('image.png')
		  );
	} catch (error) {
		// ...
	}
});
```

Reply:
I've modified the code. Should be fine now.